lib/rhc/deployment_helpers.rb

```diff
@@ -9,15 +9,17 @@ module DeploymentHelpers
 
     protected
 
-      def deploy_artifact(rest_app, artifact, hot_deploy, force_clean_build)
+      def deploy_artifact(rest_app, artifact, options)
         is_file = File.file?(artifact)
         is_url = URI::ABS_URI.match(artifact).present?
 
+        ssh_executable = check_ssh_executable! options.ssh
```

Comment:
I'd kind of rather push this down into `deploy_local_file`, `deploy_file_from_url`, and `deploy_git_ref` to avoid lengthening an already-long parameter list and to keep all the SSH-related logic together in those methods.